packages/validators/src/utils/common.js

```diff
@@ -44,3 +44,9 @@ export function isTruthy (prop) {
 export function isPromise (object) {
   return isObject(object) && isFunction(object.then)
 }
+
+export function withAsync (validator) {
+  const normalized = normalizeValidatorObject(validator)
+  normalized.$async = true
```

Comment:
This mutates the passed object, which bothers me a bit :/

Reply:
Yup! I had some issues with that in other helpers as it turned out to mutate built in validators... We might want to do a spread merge.

Reply:
I will reflect this in a new PR.